### ISSUE TITLE
Is there a way to add my own hooks for certain events?

### ISSUE THREAD

**Issue by Taytay** [hubot-scripts/hubot-github-repo-event-notifier#23]
I'd like to be able to respond to these events with some scripts of my own. From what I can tell, there isn't a way to add my own handler methods, is there?

**Comment by parkr** [comment]
[Not presently, no.](https://github.com/hubot-scripts/hubot-github-repo-event-notifier/blob/master/src/github-repo-event-notifier.coffee#L66-L69)

What kind of API would you envision?

**Comment by patcon** [comment]
Hey @Taytay! Beat me to it. Couldn't we [emit](http://nodejs.org/api/events.html#events_emitter_emit_event_arg1_arg2) a `github-event-received` action with an event type arg and let people set up listeners?

**Comment by Taytay** [comment]
You could indeed! ;) In fact, that's what I'm doing now in my local fork:

```coffeescript
      if eventType in eventTypes
        # Emit the event for other scripts to handle
        robot.emit "github-repo-event", {
            data        : data,
            eventType   : eventType
            room : room
        }
        announceRepoEvent data, eventType, (what) ->
          robot.messageRoom room, what
```

It's been easy to work with so far.

**Comment by Taytay** [comment]
Oh, and thanks for this project! Good stuff! :)

**Comment by niklasR** [comment]
@Taytay @parkr - would it be worth opening a PR for the `robot.emit`?

My use would be to automatically create a new JIRA ticket when an issue is raised, and since hubot already knows about all this, it'd be very handy to make use of it.

**Comment by patcon** [comment]
Yeah, go for it! but please provide tests so we can merge easily :)
